tests/WorkflowTest.php

```diff
@@ -36,6 +38,43 @@
     assertEquals(1, $workflow->refresh()->jobs_processed);
 });
 
+it('it stores finished job id, defaulting to class name', function () {
+    $job1 = new TestJob1();
+    $workflow = createWorkflow([
+        'job_count' => 1,
+        'jobs_processed' => 0,
+    ]);
+
+    $workflow->onStepFinished($job1);
+
+    assertEquals([$job1::class], $workflow->refresh()->finished_jobs);
+});
+
+it('it stores finished job id', function () {
+    $job1 = (new TestJob1())->withJobId('::job-id::');
+    $workflow = createWorkflow([
+        'job_count' => 1,
+        'jobs_processed' => 0,
+    ]);
+
+    $workflow->onStepFinished($job1);
+
+    assertEquals(['::job-id::'], $workflow->refresh()->finished_jobs);
+});
+
+it('it stores finished job id for nested workflow jobs', function () {
+    $workflow = new WorkflowWithWorkflow(new NestedWorkflow(
+        $job = new TestJob1()
+    ));
+    $definition = $workflow->definition();
+    [$model, $initial] = $definition->build();
+
+    $model->onStepFinished($job);
+
+    assertEquals(NestedWorkflow::class . '.' . TestJob1::class, $job->jobId);
```

Comment:
I think this assertion is unnecessary because you also added another test somewhere that specifically tests for this (ids of nested jobs get namespaced properly)